### ISSUE TITLE
Improvements to Effect Size (Eta2, Omega2) Indices for ANOVAs

### ISSUE THREAD

**Issue by DominiqueMakowski** [easystats/parameters#110]
Combining several issues here:

- [ ] Support repeated measures ANOVAs (#84, #100):
    - make it work for `anova(themixedmodel)`
    - Make sure these are *correct*
- [ ] Generalized eta2 and Omega2 (#91)
    - https://www.frontiersin.org/articles/10.3389/fpsyg.2013.00863/full
- [ ] Adjusted eta2
    - https://journals.sagepub.com/doi/abs/10.1177/2515245919855053?journalCode=ampa

Does anybody know an expert of ANOVAs that could help us going through this?

**Comment by IndrajeetPatil** [comment]
@DominiqueMakowski Maybe these functions can also be moved to the new `effectsize` package?

**Comment by DominiqueMakowski** [comment]
I don't see why not!

**Comment by mattansb** [comment]
I agree - `effectsize` seems like the perfect place for all `r`, `d`, `OR`, and `vaf` measures!

**Comment by DominiqueMakowski** [comment]
I'll need help 🙉

**Comment by IndrajeetPatil** [comment]
Until these generalized functions are implemented in `effectsize` itself, we can probably just move (literally copy-paste) the eta- and partial-omega squared functions to `effectsize` and re-export them in `parameters`? 
(Of course, this means that `effectsize` will have to go on `CRAN` soon, so this can wait if there is a new version of `parameters` in the offing.)

All the new development for these new generalized functions can then happen in `effectsize` itself.

**Comment by DominiqueMakowski** [comment]
Yup, I just wonder whether the eta2 & co functions need stuff from parameters that would need to be transferred too. 

For instance in parameters a line added the SE to the object:

https://github.com/easystats/effectsize/blob/28fa79ca7486b51b33aa4ce9c72992e1a3a95a18/R/standardize_parameters.R#L145

But I had to deactivate it because it reaquires `standard_error`, which is parameters... Not sure how to proceed in such cases. Do we move `standard_error` as well? But standard_error relies in some cases on the whole model_parameters. So we need to find a better way...

**Comment by mattansb** [comment]
Perhaps under `effectsize` some of these functions would be stripped down a bit (e.g., `.standardize_parameters_posthoc` wouldn't return standardized SEs?).

Additionally, some functions can appear, maybe internally / not-exported, in more than 1 package, if need be (obviously this should be avoided as much as possible).

**Comment by DominiqueMakowski** [comment]
> Additionally, some functions can appear, maybe internally / not-exported, in more than 1 package, if need be (obviously this should be avoided as much as possible).

this conflicts a lot with my "neatness" obsession ^^. more importantly, it becomes hard to maintain and might reflect a clunky organisation (and we are well organized aren't we) . But Maybe standard_error could be moved to insight as it is mainly a "retrieval" function?

**Comment by strengejacke** [comment]
SE for "classic" standardized parameters is useful for calculating the related CI's around the standardized parameters. That's why I added that code lines...

**Comment by strengejacke** [comment]
Why do we need standardization methods for effect sizes? Else, can't we just make parameters a dependency of effectsize?

**Comment by DominiqueMakowski** [comment]
> SE for "classic" standardized parameters is useful for calculating the related CI's around the standardized parameters. That's why I added that code lines...

I know, I know, and I believe it is indeed super useful. That's why we need to find a way to make it work

> Why do we need standardization methods for effect sizes

Some indices effect sizes are related to standardization

> Else, can't we just make parameters a dependency of effectsize?

it's more natural if effectsize is a dep of parameters

I'll think about a solution

**Comment by DominiqueMakowski** [comment]
after looking a bit more at the code for SE, eta2 omega2 etc. etc. they are really intertwined with the core of the parameters package (relying on the `extract_parameters_*` internals. Seems hard to move them from here indeed...

Simplest way is, indeed, that parameters becomes a deps of effectsize... but will it be possible then to easily have effectsize imported in parameters? I.e., circular dependencies?

**Comment by IndrajeetPatil** [comment]
Circular dependencies are not allowed by CRAN, so that’s definitely not an option. 

I think the best current solution is for effectsize to rely on parameters.

**Comment by DominiqueMakowski** [comment]
I agree that parameters becoming a dep of effectsize is problematic (as the former needs to pull effect sizes from effectsize), especially in the current state as parameters is a mid-to-high level package. 

One solution would indeed to lower the level of parameters, to re-scope it on functions surrounding parameters extraction and computation.

We could have parameter and performance in the middle of the hierarchy and effectsize on the side, providing functions to complete and/or interpret the output of both. On top of that, report (which is being more and more stripped down of features) would finally find its role as the integrator: it would run model_parameters (in parameters), model_performance (in performance), interpret_parameters and interpret_performance (to be implemented in effectsize) and would merge it all it in a nicely formatted table. Et voilà. 

(Side note, maybe if we move some stuff (feature reduction) to correlation it might good to rename it, maybe something like "multivariate" or something that would encompass these kind of things.)

**Comment by strengejacke** [comment]
Do you want parameters as suggested package by effectsize, so we still can compute SE from classic standardized parameters?

**Comment by DominiqueMakowski** [comment]
parameters will be imported by effectsize, so yeah we will have access to everything in the longrun.

So, currently effectsize is stable, i.e., it passes tests and has some basic features. I think our top priority should be to re-stabilize the ecosystem, has things were broken due to these changes (for instance I think report doesn't work)

Thus, we could 1) submit the stripped parameters (without standardization) and 2) submit effectsize (so the new organization is "there" and we can adapt the remaining packages to it), and work on cleanly transferring the ANOVA Effect size indices for the next updates (as we need to address the issue of extract_parameters for that). How does that sound?

**Comment by strengejacke** [comment]
There are problems with reverse dependencies, in particular sjPlot, which uses standardization from parameters. So once effectsize is on CRAN, parameters should at least re-export those functions.

**Comment by DominiqueMakowski** [comment]
mmh but currently it is the contrary, I. e.,effectsize is not a dep of parameters (because the contrary is necessary). best would be to pull these functions directly from effectsize once it is out. it is rather a contained and hopefully stable package (by the time it is out)

**Comment by mattansb** [comment]
I somewhat agree with Daniel: moving functions around can break any outside
dependancy on `parameters`.
We should address this somehow - either by re-exporting functions, or by
giving explicit errors in parameters for a while:

```r
parameters::foo <- function (...){
  warning ("this function is on 'effextsize'...")
  effectsize::too(...)
}
```

Or something like that...

--
Mattan S. Ben-Shachar, PhD student
Department of Psychology & Zlotowski Center for Neuroscience
Ben-Gurion University of the Negev
The Developmental ERP Lab

**Comment by mattansb** [comment]
I somewhat agree with Daniel: moving functions around can break any outside
dependency on `parameters`.
We should address this somehow - either by re-exporting functions, or by
giving explicit errors in parameters for a while:

```r
parameters::foo <- function (...){
  warning ("this function is on 'effextsize'...")
  effectsize::too(...)
}
```

Or something like that...

**Comment by strengejacke** [comment]
Maybe softly deprecate? First, re-export, then `.Deprecate()` and finally `.Defunct()`...

**Comment by mattansb** [comment]
👆
